pkg/k8shandler/kibana/reconciler.go

```diff
@@ -0,0 +1,533 @@
+package kibana
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"reflect"
+
+	configv1 "github.com/openshift/api/config/v1"
+	kibana "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
+	"github.com/openshift/elasticsearch-operator/pkg/constants"
+	"github.com/openshift/elasticsearch-operator/pkg/utils"
+	"github.com/sirupsen/logrus"
+	apps "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/util/retry"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	kibanaServiceAccountName     = "kibana"
+	kibanaOAuthRedirectReference = "{\"kind\":\"OAuthRedirectReference\",\"apiVersion\":\"v1\",\"reference\":{\"kind\":\"Route\",\"name\":\"kibana\"}}"
+)
+
+var (
+	kibanaServiceAccountAnnotations = map[string]string{
+		"serviceaccounts.openshift.io/oauth-redirectreference.first": kibanaOAuthRedirectReference,
```

Comment:
unnecessary?

Reply:
@jcantrill are we still use this annotation? https://github.com/igor-karpukhin/elasticsearch-operator/blob/faf21d26df43f541631a7e14dcbe93158ce24e09/pkg/k8shandler/kibana/serviceaccount.go#L34

Reply:
^ to specify, does this annotation replace the use of `oauthclient` objects?

Reply:
yes. https://github.com/openshift/cluster-logging-operator/pull/234